### PR TITLE
Differentiate CONNECT requests from other requests

### PR DESCRIPTION
**Pull request by kkkkun**
Signed-off-by: kkkkun <scuzk373x@gmail.com>

#### What type of PR is this?
/kind bug

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/pull/81660#issuecomment-1214569062

Log pod, exec pod and portforward pod are longRunning requests.  We should record these requests  to CONNECT requests. 

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go [hunk 1]
```diff
@@ -288,6 +288,14 @@ var (
 		"UPDATE",
 		"WATCH",
 		"WATCHLIST")
+
+	//these are the valid connect request which we report in our metrics.
```
Comment (yangjunmyfm192085) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:292]:
```suggestion
	// these are the valid connect requests which we report in our metrics.
```

Comment (kkkkun) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: Thank you.

**Comment by wojtek-t** [comment]
/ok-to-test

@kkkkun - can you please rebase this PR? It's a valid change...

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics_test.go [hunk 2]
```diff
@@ -142,14 +143,79 @@ func TestCleanVerb(t *testing.T) {
...
+				},
+			},
+			requestInfo: &request.RequestInfo{
+				Verb:              "POST",
+				Resource:          "pods",
+				IsResourceRequest: true,
+				Subresource:       "portforward",
+			},
+			expectedVerb: "CONNECT",
+		},
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics_test.go:210]: Please add a testscale where subresource is not in a `validConnectRequests`

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go [hunk 3]
```diff
@@ -427,7 +435,7 @@ func RecordRequestAbort(req *http.Request, requestInfo *request.RequestInfo) {
 	}
 
 	scope := CleanScope(requestInfo)
-	reportedVerb := cleanVerb(CanonicalVerb(strings.ToUpper(req.Method), scope), getVerbIfWatch(req), req)
+	reportedVerb := cleanVerb(CanonicalVerb(strings.ToUpper(req.Method), scope), getVerbIfWatch(req), req, requestInfo)
```
[the remark this answers is not in the document]
Comment (kkkkun) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:438]: That's OK.

[the remark this answers is not in the document]
Comment (kkkkun) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: Rebase done!

**Comment by dims** [comment]
If you still need this PR then please rebase, if not, please close the PR

**Comment by kkkkun** [comment]
ping @wojtek-t

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go [hunk 4]
```diff
@@ -321,6 +321,14 @@ var (
 		"UPDATE",
 		"WATCH",
 		"WATCHLIST")
+
+	//these are the valid connect request which we report in our metrics.
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:325]: // These are the valid connect requests which we report in our metrics.

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go [hunk 5]
```diff
@@ -321,6 +321,14 @@ var (
...
+	validConnectRequest = utilsets.NewString(
```
Comment (wojtek-t) [inline staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go:326]: nit: validConnectRequests

**Review by wojtek-t — COMMENTED** [review]
Just two very minor nits - other than that LGTM

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go (hunk 5 above)
Comment (kkkkun) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: Thanks.
Fixed！

File: staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go (hunk 4 above)
Comment (kkkkun) [reply staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go]: Thanks.
Fixed！

**Pull request merged**